cloudant-client/src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -638,41 +642,40 @@ public static ClientBuilder bluemix() {
      * service instance.
      * <P>
      * Note: Specifying a service name is only required when multiple Cloudant services are
-     * available. If there is only a single bound Cloudant service instance then {@link #bluemix()}
-     * can also be used.
+     * available. If there is only a single bound Cloudant service instance then
+     * {@link #bluemix(String)} can also be used.
      * </P>
      *
-     * @param serviceName name of Bluemix service instance or {@code null} to try to use the only
+     * @param vcapServices contents of VCAP_SERVICES environment variable
+     * @param instanceName name of Bluemix service instance or {@code null} to try to use the only
      *                    available Cloudant service
      * @return a new ClientBuilder for the account
      *
      * @throws IllegalArgumentException if the Cloudant service instance is not available
-     * @throws IllegalStateException    if the VCAP_SERVICES environment variable is invalid or missing
      */
-    public static ClientBuilder bluemix(String serviceName) {
+    public static ClientBuilder bluemix(String vcapServices, String instanceName) {
         Gson gson = new GsonBuilder().create();
 
-        String vcap_env = System.getenv("VCAP_SERVICES");
-        if (vcap_env == null) {
-            throw new IllegalStateException("VCAP_SERVICES is not present.");
+        if (vcapServices == null) {
+            throw new IllegalArgumentException("VCAP_SERVICES is not present.");
         }
 
         CloudFoundryServices services;
         try {
-            services = gson.fromJson(vcap_env, CloudFoundryServices.class);
+            services = gson.fromJson(vcapServices, CloudFoundryServices.class);
         } catch (JsonParseException e){
-            throw new IllegalStateException("VCAP_SERVICES is not valid JSON.", e);
+            throw new IllegalArgumentException("VCAP_SERVICES is not valid JSON.", e);
```

Comment:
If we make the above change then we should be consistent everywhere and this should be:
The vcapServices JSON information is not valid JSON.

Although that's not the smoothest message I've ever read!